### PR TITLE
Split text.php to Item class

### PR DESCRIPTION
**Pull request by zeroadam**
move item related functions from text.php to src/Model/Item class
Related to: #3878

### CODE REVIEW

**Comment by zeroadam** [comment]
I ran this one overnight. No errors to speak of. 

I couldn't tell really, if there were more functions that should be moved here as well.

**Comment by zeroadam** [comment]
hmm.. actually, looks like get_plink() could be moved as well

File: src/Model/Item.php [hunk 1]
```diff
@@ -3247,4 +3253,295 @@ public static function getPermissionsSQLByUserId($owner_id, $remote_verified = f
...
+	 * @param array $item
+	 * @return boolean|array False if item has not plink, otherwise array('href'=>plink url, 'title'=>translated title)
+	 */
+	public static function getPlink($item)
+	{
+		$a = self::getApp();
+
+		if ($a->user['nickname'] != "") {
+			$ret = [
+					'href' => "display/" . $item['guid'],
```
Comment (MrPetovan) [inline src/Model/Item.php]: Please unindent this block 1 level.

File: src/Model/Item.php [hunk 2]
```diff
@@ -3247,4 +3253,295 @@ public static function getPermissionsSQLByUserId($owner_id, $remote_verified = f
...
+				];
+
+			if (x($item, 'plink')) {
+				$ret["href"] = $a->removeBaseURL($item['plink']);
+				$ret["title"] = L10n::t('link to source');
+			}
+
+		} elseif (x($item, 'plink') && ($item['private'] != 1)) {
+			$ret = [
+					'href' => $item['plink'],
```
Comment (MrPetovan) [inline src/Model/Item.php]: Please unindent this block 1 level.

File: src/Model/Item.php [hunk 3]
```diff
@@ -3247,4 +3253,295 @@ public static function getPermissionsSQLByUserId($owner_id, $remote_verified = f
...
+
+		if ($a->user['nickname'] != "") {
+			$ret = [
+					'href' => "display/" . $item['guid'],
+					'orig' => "display/" . $item['guid'],
+					'title' => L10n::t('View on separate page'),
+					'orig_title' => L10n::t('view on separate page'),
+				];
+
+			if (x($item, 'plink')) {
```
Comment (MrPetovan) [inline src/Model/Item.php]: Please replace `x()` by `!empty()`

File: src/Model/Item.php [hunk 4]
```diff
@@ -3247,4 +3253,295 @@ public static function getPermissionsSQLByUserId($owner_id, $remote_verified = f
...
+				$ret["href"] = $a->removeBaseURL($item['plink']);
+				$ret["title"] = L10n::t('link to source');
+			}
+
+		} elseif (x($item, 'plink') && ($item['private'] != 1)) {
```
Comment (MrPetovan) [inline src/Model/Item.php]: Please replace `x()` by `!empty()`

File: src/Model/Item.php [hunk 5]
```diff
@@ -3247,4 +3253,295 @@ public static function getPermissionsSQLByUserId($owner_id, $remote_verified = f
...
+		while ((strpos($s, $authorsearch) !== false)) {
+			$pos = strpos($s, $authorsearch);
+			$rnd = random_string(8);
+			$authorreplace = '<br /> <span id="author-wrap-' . $rnd . '" class="author-wrap fakelink" onclick="openClose(\'author-' . $rnd . '\');">' . L10n::t('Click to open/close') . '</span>'.
+						'<blockquote class="author" id="author-' . $rnd . '" style="display: block;">';
+			$s = substr($s, 0, $pos) . $authorreplace . substr($s, $pos + strlen($authorsearch));
+		}
+
+		// Replace friendica image url size with theme preference.
+		if (x($a->theme_info, 'item_image_size')) {
```
Comment (MrPetovan) [inline src/Model/Item.php]: Please replace `x()` by `!empty()`

**Pull request merged**